### PR TITLE
SF #1859 Mail to Ticket Automation - issue with long subjects

### PR DESCRIPTION
**Pull request by jbostoen**
Fixes issue with long subjects and references (multiline in EML).
Actually might need to be applied to all header lines and not just "subject" and "references", but I had no reason to further investigate this.

### CODE REVIEW

**Comment by Hipska** [comment]
Reference: https://sourceforge.net/p/itop/tickets/1859/

File: classes/rawemailmessage.class.inc.php [hunk 1]
```diff
@@ -582,7 +582,8 @@ protected function ExtractHeadersAndRawBody($aLines)
 			{
 				if (isset($aRawFields[$sCurrentHeader]))
 				{
-					$aRawFields[$sCurrentHeader] .= substr($sLine, 1);
+					// Fix for long subjects where spaces get lost on split header lines.
+					$aRawFields[$sCurrentHeader] .= (in_array($sCurrentHeader, array('references', 'subject')) ? $sLine : substr($sLine, 1));
```
Comment (Hipska) [inline classes/rawemailmessage.class.inc.php:586]:
```suggestion
					// Fix for long values where spaces get lost on split header lines.
					$aRawFields[$sCurrentHeader] .= $sLine;
```

Comment (jbostoen) [reply classes/rawemailmessage.class.inc.php:586]: @piRGoif would this need more investigation or can we just apply it to all headers?

Comment (jbostoen) [reply classes/rawemailmessage.class.inc.php:586]: @piRGoif ?

Comment (piRGoif) [reply classes/rawemailmessage.class.inc.php:586]: Hello,
Sorry I'm already in charge of lots of PR, I cannot handle this one right now :/

**Comment by piRGoif** [comment]
This is planned for the next Mail to ticket automation release that will be included in iTop 2.7.5 professional and essential.
Combodo internal ref : N°3013

**Review by accognet — APPROVED** [review]

File: classes/rawemailmessage.class.inc.php (hunk 1 above)
Comment (Hipska) [reply classes/rawemailmessage.class.inc.php:586]: @accognet there was still an open question here..

Comment (accognet) [reply classes/rawemailmessage.class.inc.php:586]: The code _substr($sLine, 1)_ allow to remove \t characters in the beginning of some lines. I prefer to keep the PR with test on type of header, in order to better control the modification. I suppose that this _substr_ was not added by chance.

Comment (Hipska) [reply classes/rawemailmessage.class.inc.php:586]: So the best fix would actually be like this?
```suggestion
					// Fix for long subjects where spaces get lost on split header lines.
					$aRawFields[$sCurrentHeader] .= ltrim($sLine, "\t");
```

**Pull request merged**